src/java.base/share/classes/sun/nio/ch/Interruptible.java

```diff
@@ -31,6 +31,21 @@
 
 public interface Interruptible {
 
-    public void interrupt(Thread t);
+    /**
+     * Invoked by Thread.interrupt when the given Thread is interrupted. This method
+     * is invoked while holding the Thread's interrupt lock. It will typically record
+     * that the I/O operation has been interrupted so that it can be coordinated with
+     * {@code postInterrupt} when it called after releasing the Thread's interrupt
+     * lock. This method must not block.
+     */
+    void interrupt(Thread target);
+
+    /**
+     * Invoked by Thread.interrupt after releasing the Thread's interrupt lock. It
+     * may also invoked by AbstractInterruptibleChannel/AbstractSelector begin methods
+     * when the Thread is interrupted. This method closes the channel or wakes up the
+     * Selector. This method is required to be idempotent.
+     */
+    void postInterrupt();
```

Comment:
This is an internal interface, it's not a general purpose interface that anything outside of the NIO implementation should use. The phrase in the javadoc is meant to make it clear that it may be called more than once, and from several different I/O ops. on the channel.